src/auto-advance.js

```diff
@@ -1,97 +1,97 @@
+    this.fullReset();
 
-/**
- * Sets up auto-advance behavior on a player.
- *
- * @param  {Player} player
- *         the current player
- *
- * @param  {number} delay
- *         The number of seconds to wait before each auto-advance.
- *
- * @return {undefined}
- *         Used to short circuit function logic
- */
-const setup = (player, delay) => {
-  reset(player);
+    // If delay is invalid or undefined, do nothing further (auto-advance already cancelled)
+    if (typeof seconds !== 'number' || seconds < 0 || !isFinite(seconds)) {
+      return;
+    }
 
-  // Before queuing up new auto-advance behavior, check if `seconds` was
-  // called with a valid value.
-  if (!validSeconds(delay)) {
-    player.playlist.autoadvance_.delay = null;
-    return;
+    // Set the new delay and start listening for 'ended' to trigger auto-advance
+    this.delay_ = seconds;
+    this.player_.on('ended', this.startTimeout_);
   }
 
-  player.playlist.autoadvance_.delay = delay;
+  /**
+   * Gets the delay for auto-advance.
+   */
+  getDelay() {
+    return this.delay_;
+  }
 
-  player.playlist.autoadvance_.trigger = function() {
+  /**
+   * Starts the auto-advance timeout.
+   *
+   * @private
+   */
+  startTimeout_() {
+    // Ensure we don't stack timeouts
+    this.clearTimeout_();
 
-    // This calls setup again, which will reset the existing auto-advance and
-    // set up another auto-advance for the next "ended" event.
-    const cancelOnPlay = () => setup(player, delay);
+    // Set a new timeout for auto-advancing
+    if (this.delay_ !== null) {
+      // Listen for a play event to cancel auto-advance if it occurs before the timeout completes
+      this.clearTimeoutOnPlay = () => this.clearTimeout_();
+      this.player_.one('play', this.clearTimeoutOnPlay);
 
-    // If there is a "play" event while we're waiting for an auto-advance,
-    // we need to cancel the auto-advance. This could mean the user seeked
-    // back into the content or restarted the content. This is reproducible
-    // with an auto-advance > 0.
-    player.one('play', cancelOnPlay);
+      this.timeoutId_ = setTimeout(() => {
+        this.advanceCallback_();
 
-    player.playlist.autoadvance_.timeout = player.setTimeout(() => {
-      reset(player);
-      player.off('play', cancelOnPlay);
-      player.playlist.next();
-    }, delay * 1000);
-  };
+        // Clean up the listener for the play event when the auto-advance triggers
+        this.player_.off('play', this.clearTimeoutOnPlay);
+      }, this.delay_ * 1000);
+    }
+  }
 
-  player.one('ended', player.playlist.autoadvance_.trigger);
-};
+  /**
+   * Clears the current auto-advance timeout and removes the 'play' event listener.
+   *
+   * @private
+   */
+  clearTimeout_() {
+    if (this.timeoutId_) {
+      clearTimeout(this.timeoutId_);
+      this.timeoutId_ = null;
+      this.player_.off('play', this.clearTimeoutOnPlay);
+    }
+  }
```

Comment:
```suggestion
  clearTimeout_ = () => {
    if (this.timeoutId_) {
      clearTimeout(this.timeoutId_);
      this.timeoutId_ = null;
      this.player_.off('play', this.clearTimeoutOnPlay);
    }
  }
```

You can auto-bind this method, so you don't have to create wrappers like `this.clearTimeoutOnPlay`